core/src/apps/management/ble/pair_new_device.py

```diff
@@ -50,12 +50,22 @@ async def pair_new_device() -> None:
                 ),
                 None,
             )
+            if result is not CONFIRMED:
+                raise ActionCancelled
         except Exception:
             ble.reject_pairing()
             raise
         else:
-            if result is CONFIRMED:
-                ble.allow_pairing(code)
+            ble.allow_pairing(code)
+            try:
+                # wait for the host code confirmation
+                result = await interact(trezorui_api.wait_ble_host_confirmation(), None)
+                if result is CANCELLED:
```

Comment:
Btw by default `interact` raises ActionCancelled if the layout returns CANCELLED so this never gets executed.